### PR TITLE
Update routing samples

### PR DESCRIPTION
**Pull request by DullReferenceException**
Updated router samples to use the EventEmitter

### CODE REVIEW

File: examples/with-google-analytics/components/Page.js [hunk 1]
```diff
@@ -4,9 +4,9 @@ import Header from './Header'
 
 import * as gtag from '../lib/gtag'
 
-Router.onRouteChangeComplete = url => {
-  gtag.pageview(url)
-}
+Router.ready(() => {
+  Router.router.events.on('routeChangeComplete', url => gtag.pageview(url))
```
Comment (timneutkens) [inline examples/with-google-analytics/components/Page.js]: Shouldn't this be `Router.events.on` (based on your change on canary)

Comment (timneutkens) [reply examples/with-google-analytics/components/Page.js]: Basically I don't want to promote the use of the internal property if we don't have to 👍

Comment (DullReferenceException) [reply examples/with-google-analytics/components/Page.js]: I think having `events` on the static router would be great idea, but my change didn't do that. It's only attached to the public router instance attached to the child context at this time. Will see about adding that.

Comment (DullReferenceException) [reply examples/with-google-analytics/components/Page.js]: You're right. I guess the sample isn't installing the canary version, which is throwing me off.

**Review by timneutkens — APPROVED** [review]
Awesome, I'll merge after canary is released 👍

File: examples/with-loading/components/Header.js [hunk 2]
```diff
@@ -4,12 +4,15 @@ import Link from 'next/link'
...
-Router.onRouteChangeComplete = () => NProgress.done()
-Router.onRouteChangeError = () => NProgress.done()
+Router.ready(() => {
+  const events = Router.events
+  events.on('routeChangeStart', (url) => {
+    console.log(`Loading: ${url}`)
+    NProgress.start()
+  })
+  events.on('routeChangeComplete', () => NProgress.done())
+  events.on('routeChangeError', () => NProgress.done())
```
Comment (kachkaev) [inline examples/with-loading/components/Header.js]: Why not use Router.events three times? IMO a new var creates more complexity than helps.

File: examples/with-google-analytics/components/Page.js [hunk 3]
```diff
@@ -4,9 +4,9 @@ import Header from './Header'
 
 import * as gtag from '../lib/gtag'
 
-Router.onRouteChangeComplete = url => {
-  gtag.pageview(url)
-}
+Router.ready(() => {
+  Router.events.on('routeChangeComplete', url => gtag.pageview(url))
+})
```
Comment (kachkaev) [inline examples/with-google-analytics/components/Page.js]: Will Router.ready work well during SSR (ie simply never get called)?

File: examples/with-loading/components/Header.js (hunk 2 above)
Comment (DullReferenceException) [reply examples/with-loading/components/Header.js]: Agreed, updated sample.

File: examples/with-google-analytics/components/Page.js (hunk 3 above)
Comment (DullReferenceException) [reply examples/with-google-analytics/components/Page.js]: Yep: https://github.com/zeit/next.js/blob/canary/lib/router/index.js#L10-L12

**Comment by DullReferenceException** [comment]
Simplified to use the changes introduced by https://github.com/zeit/next.js/pull/4874

**Comment by timneutkens** [comment]
Could you upgrade to the latest canary, then we can merge 👍

**Comment by DullReferenceException** [comment]
@timneutkens: should be good to go.

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Very nice! 💯 Thanks @DullReferenceException!

**Comment by kachkaev** [comment]
@DullReferenceException it seems like there's one more example left, according to github search:

  🔎 [onRouteChangeStart](https://github.com/zeit/next.js/search?q=onRouteChangeStart)  
  🔎 [onRouteChangeComplete](https://github.com/zeit/next.js/search?q=onRouteChangeComplete)  
  🔎 [onRouteChangeError](https://github.com/zeit/next.js/search?q=onRouteChangeError)

Once [with-flow](https://github.com/zeit/next.js/tree/canary/examples/with-flow) is updated, https://github.com/zeit/next.js/pull/4864 can be considered as closed. Would you be happy to submit a PR?

**Comment by DullReferenceException** [comment]
Those flow definitions are unused by the `with-flow` sample code, so should we just remove them? PR here: https://github.com/zeit/next.js/pull/5094

**Pull request merged**